### PR TITLE
Add support for pulling down dependencies via curl

### PR DESCRIPTION
**Pull request by hectcastro**
OS X doesn't have `wget` installed by default. Despite the fact that most of us have figured out a way to install it, I'd like to add support for `curl` so that we can make one less inconvenience for those who don't have `wget` installed.

### CODE REVIEW

File: tools/build-solr.sh [hunk 1]
```diff
@@ -97,8 +97,10 @@ else
     SOLR_FILE=$(basename $URL)
     SOLR_DIR=${SOLR_FILE%-src.tgz}
 
-    if test ! -e $SOLR_FILE; then
+    if which wget > /dev/null && test ! -e $SOLR_FILE; then
```
Comment (macintux) [inline tools/build-solr.sh]: Should `download()` be copied to this script as well, or at least use the same `curl` and `wget` progress arguments?

Comment (hectcastro) [inline tools/build-solr.sh]: I'm not opposed to making each download attempt uniform. I'll defer the decision to @rzezeski.

Comment (coderoshi) [inline tools/build-solr.sh]: @hectcastro if you're in here, may as well extract this to a download func

**Comment by hectcastro** [comment]
@coderoshi @macintux I pulled `download()` into `build-solr.sh`.

**Comment by coderoshi** [comment]
+1 201ac6c

**Comment by hectcastro** [comment]
@borshop merge

**Pull request merged**